### PR TITLE
Implement previous, next, first and last frame buttons

### LINKED ISSUE #141: Additional playback controls
As well as eventual keyboard shortcuts there should be buttons in the interface for previewing the previous and next frame on the framereel as well as buttons for jumping to the first and last frame captured. (basically buttons for `|<`, `<`, `>` and `>|`)

A play from start function should exist. This wouldn't have a button in the interface but could be activated by shortcut `p` or `numpad .`

### PR DESCRIPTION
**Pull request by BoatsAreRockable**
This PR fixes #141 by implementing previous, next, first and last frame buttons. In addition it moves the audio toggle from the sidebar to the playback controls area:

issue-141

From the picture above you can see that this does make the playback controls area a bit cluttered. As a result, when #155 is implemented, I think it would be good if the "next" and "previous" buttons were removed and their functionality be made available purely via keyboard shortcuts.

Playback controls area without previous and next buttons:
issue-141-2

### CODE REVIEW

**Comment by le717** [comment]
Does this need to be merged before #136 is implemented?

**Comment by BoatsAreRockable** [comment]
@le717 Ah that was not something I had considered. I suppose so.

**Comment by le717** [comment]
In terms of behavior and UI, it works and it seems you have handled all the cases. I'll look at the code part right after dinner.

TBH, I would not have moved the audio toggle there. I've been meaning to file this issue, but it should just be moved into the Capture menu. It does not need to be present in the UI.

**Comment by BoatsAreRockable** [comment]
> I would not have moved the audio toggle there.
> It does not need to be present in the UI.

@le717 Good point. I'll move it.

File: app/js/main.js [hunk 1]
```diff
@@ -543,7 +528,7 @@ function _toggleOnionSkin(ev) {
  */
 function audio(name) {
     "use strict";
-    if (btnAudioToggle.classList.contains("active")) {
+    if (playAudio === true) {
```
Comment (le717) [inline app/js/main.js]: Should be a truthy check:

`if (playAudio) {`

File: app/js/main.js [hunk 2]
```diff
@@ -696,7 +681,6 @@ function previewCapturedFrames() {
     // Begin playing the frames
     isPlaying = true;
     playBackLoop = setInterval(_videoPlay, (1000 / frameRate));
-    console.info("Playback started");
```
Comment (le717) [inline app/js/main.js]: Why was this message removed?

File: app/js/main.js [hunk 3]
```diff
@@ -46,11 +46,15 @@ var width  = 640,
     btnLoop       = document.querySelector("#btn-loop"),
     playback      = document.querySelector("#playback"),
     btnPlayPause  = document.querySelector("#btn-play-pause"),
+    btnNext       = document.querySelector("#btn-next"),
```
Comment (le717) [inline app/js/main.js]: Change ID selectors to following format for clarity:

`#btn-frame-[location]` (e.g. `#btn-frame-next`)

File: app/js/main.js [hunk 4]
```diff
@@ -244,13 +248,51 @@ function startup() {
...
+        btnLiveView.click();
+      }
+    }
+  });
+
+  // Preview one frame to the left on framereel
+  btnPrevious.addEventListener("click", function() {
+    if (curSelectedFrame > 1) {
+        videoJumpTo(curSelectedFrame - 1);
+    } else  if (winMode === "capture") {
```
Comment (le717) [inline app/js/main.js]: Nit: remove extra space between else and if.

File: app/js/main.js (hunk 3 above)
Comment (le717) [inline app/js/main.js]: Do the same the variable names too (e.g., `btnFrameNext`)

File: app/js/main.js [hunk 5]
```diff
@@ -484,7 +528,7 @@ function _toggleOnionSkin(ev) {
  */
 function audio(name) {
     "use strict";
-    if (audioToggle.checked) {
+    if (playAudio === true) {
```
Comment (le717) [inline app/js/main.js]: Same truthy check here.

File: app/js/main.js [hunk 6]
```diff
@@ -563,15 +607,31 @@ function videoPause() {
...
+  curPlayFrame = 0;
+  console.info("Playback stopped");
+}
+
+/**
+ * Pause playback and view a specific frame in the preview area.
+ *
+ * @param {Integer} id The frame ID to preview.
+ */
+function videoJumpTo(id) {
```
Comment (le717) [inline app/js/main.js]: Try `function _displayFrame(id) {` for clarity.

**Comment by le717** [comment]
I've gone though all I can ATM, will review more later. Fix up those changes for now. ;)

File: app/js/main.js (hunk 2 above)
Comment (BoatsAreRockable) [inline app/js/main.js]: @le717 Ah this is was an accident!

File: app/js/main.js [hunk 7]
```diff
@@ -964,6 +1023,20 @@ function loadMenu() {
       modifiers: "ctrl",
     }));
 
+  captureMenu.append(new nw.MenuItem({ type: "separator" }));
+
+  captureMenu.append(new nw.MenuItem({
+    label: "Play capture sounds",
+    click: function() {
+      playAudio = !playAudio;
+      notifyInfo(`Capture sounds ${playAudio ? "enabled" : "disabled"}`);
```
Comment (le717) [inline app/js/main.js]: I don't think we need this line. The checkbox in the Capture menu is enough indication.

**Comment by le717** [comment]
Alright, review done. I like it. When you finish up the review changes, let me know and I'll merge it.

BTW, I plan on fixing the playback as the next change, then keyboard shortcuts, then we'll call it v0.7.1 and you can promote it on BiM. We can't mere

File: app/js/main.js (hunk 7 above)
Comment (BoatsAreRockable) [inline app/js/main.js]: @le717 The only thing about removing this is that it means there is visual response when this called via `ctrl+m`.

**Comment by BoatsAreRockable** [comment]
@le717 Thank you for reviewing this, I've just finished making the review changes.

The v0.7.1 plan sounds good. You may have noticed that I haven't really promoted 0.7.0 since it didn't bring any major visual changes (and because of the playback issue) but 0.7.1 will get the full blog and BiM treatment!

File: app/js/main.js (hunk 7 above)
Comment (le717) [inline app/js/main.js]: In that case, its fine. Multiple notification support will help things.

**Pull request merged**